### PR TITLE
data convertor

### PR DESCRIPTION
**Pull request by sstevenshang**
**Don't merge!**

I've written the data convertor class, which parses the .map and .ped files and output to either an output file or a DenseFileData object.

Please take a look, but not ready for merge yet, since we still need to:
1. Integrate into SPAEML so it can directly use the DenseFileData object.
2. Write unit tests.

### CODE REVIEW

**Review by jacobrh91 — CHANGES_REQUESTED** [review]
This looks great! A few small points:

In general, let's refer to the new file parser as PedMapParser instead of PLINKParser, since PLINK has other formats (like .bgen) that we may also parse in time.

ConvertFormat:

* Why treat the input files in the OptionParser as Strings instead of files? No problem with this, I just want to know why
* Instead of using a regex to determine if the suffix on the files are .ped/.map, just check the last 4 characters with something like: filter (x => x.takeRight(4) == .ped/.map). I think this looks a bit cleaner (and may be more efficient, not that that matters here very much)

PlinkFileParser
* Your documentation on the .ped/.map format is great! Terse but still really clear! I spotted one problem though. The individual ID is the one that is unique, not the family ID.
* On that note, on line 58, it looks like you're concatenating the family ID and the individual ID and treating that as the sampleName. Why do this (or am I misunderstanding something)?
* The logic in getAllMajorAndMinorAlleles is pretty hard to follow. Some comment lines would help a lot.
    * Not sure if this logic is correct. You appear to be concatenating the two alleles ("A" "T" becomes "AT", for example) and then count the frequency of that pair in allelesCount. But the major/minor allele only refers to the count of each individual letter (if A is present most, it is the major allele; the frequency of particular pairs do not matter, if my understanding is correct)
* I believe the logic of encodeSamplesForAllSNPs is good.

Thanks,
Jacob

**Comment by sstevenshang** [comment]
PlinkFileParser
- My understanding is that the family IDs are unique (no two families have the same ID) and the individual IDs are unique within the same family. Hence, family ID concatenated with individual ID would guarantee an unique global ID across entire sample population.
- The `++` operator is for concatenating lists, not elements. So [A, T] ++ [A, A] would = [A, T, A, A]. 
- I will add more documentations inside the function.

**Comment by sstevenshang** [comment]
ConvertFormat
- It was just easier to get a list of String then a list of Files.
- Good points, takeRight is a lot better.

**Comment by sstevenshang** [comment]
PED/MAP parse is now integrated into SPAEML, which can be run with: `java -jar target/SPAEML-0.0.1-jar-with-dependencies.jar StepwiseModelSelection --ped [.ped file] --map [.map file] -P [phenotype file] -o [output dir]`.

Waiting on data for testing.

**Review by jacobrh91 — APPROVED** [review]
